vendor/windows_game_input/windows_game_input.odin

```diff
@@ -0,0 +1,1129 @@
+	NamedArrayItemCollection = 4,
+	UsageSwitchItemCollection = 5,
+	UsageModifierItemCollection = 6
+}
+
+GameInputRawDevicePhysicalUnitKind :: enum c.int {
+	Unknown = -1,
+	None = 0,
+	Time = 1,
+	Frequency = 2,
+	Length = 3,
+	Velocity = 4,
+	Acceleration = 5,
+	Mass = 6,
+	Momentum = 7,
+	Force = 8,
+	Pressure = 9,
+	Angle = 10,
+	AngularVelocity = 11,
+	AngularAcceleration = 12,
+	AngularMass = 13,
+	AngularMomentum = 14,
+	AngularTorque = 15,
+	ElectricCurrent = 16,
+	ElectricCharge = 17,
+	ElectricPotential = 18,
+	Energy = 19,
+	Power = 20,
+	Temperature = 21,
+	LuminousIntensity = 22,
+	LuminousFlux = 23,
+	Illuminance = 24
+}
+
+GameInputRawDeviceReportItemFlag :: enum c.int {
+	ConstantItem = 0,
+	ArrayItem = 1,
+	RelativeItem = 2,
+	WraparoundItem = 3,
+	NonlinearItem = 4,
+	StableItem = 5,
+	NullableItem = 6,
+	VolatileItem = 7,
+	BufferedItem = 8
+}
+GameInputRawDeviceReportItemFlags :: bit_set[GameInputRawDeviceReportItemFlag; c.int]
+
+GameInputRawDeviceReportKind :: enum c.int {
+	InputReport = 0,
+	OutputReport = 1,
+	FeatureReport = 2
+}
+
+GameInputRumbleMotors_Flag :: enum c.int {
+	LowFrequency = 0,
+	HighFrequency = 1,
+	LeftTrigger = 2,
+	RightTrigger = 3
+}
+GameInputRumbleMotors :: bit_set[GameInputRumbleMotors_Flag; c.int]
+
+GameInputSwitchKind :: enum c.int {
+	UnknownSwitchKind = -1,
+	GameInput2WaySwitch = 0,
+	GameInput4WaySwitch = 1,
+	GameInput8WaySwitch = 2
+}
+
+GameInputSwitchPosition :: enum c.int {
+	Center = 0,
+	Up = 1,
+	UpRight = 2,
+	Right = 3,
+	DownRight = 4,
+	Down = 5,
+	DownLeft = 6,
+	Left = 7,
+	UpLeft = 8
+}
+
+GameInputSystemButtons_Flag :: enum c.int {
+	Guide = 0,
+	Share = 1 
+}
+GameInputSystemButtons :: bit_set[GameInputSystemButtons_Flag; c.int]
+
+GameInputTouchShape :: enum c.int {
+	Unknown = -1,
+	Point = 0,
+	Shape1DLinear = 1,
+	Shape1DRadial = 2,
+	Shape1DIrregular = 3,
+	Shape2DRectangular = 4,
+	Shape2DElliptical = 5,
+	Shape2DIrregular = 6
+}
+
+GameInputUiNavigationButtons_Flag :: enum c.int {
+	Menu = 0,
+	View = 1,
+	Accept = 2,
+	Cancel = 3,
+	Up = 4,
+	Down = 5,
+	Left = 6,
+	Right = 7,
+	Context1 = 8,
+	Context2 = 9,
+	Context3 = 10,
+	Context4 = 11,
+	PageUp = 12,
+	PageDown = 13,
+	PageLeft = 14,
+	PageRight = 15,
+	ScrollUp = 16,
+	ScrollDown = 17,
+	ScrollLeft = 18,
+	ScrollRight = 19
+}
+GameInputUiNavigationButtons :: bit_set[GameInputUiNavigationButtons_Flag; c.int]
+
+// Structs
+
+APP_LOCAL_DEVICE_ID :: distinct [32]byte
+
+GameInputArcadeStickInfo :: struct {
+	menuButtonLabel: GameInputLabel,
+	viewButtonLabel: GameInputLabel,
+	stickUpLabel: GameInputLabel,
+	stickDownLabel: GameInputLabel,
+	stickLeftLabel: GameInputLabel,
+	stickRightLabel: GameInputLabel,
+	actionButton1Label: GameInputLabel,
+	actionButton2Label: GameInputLabel,
+	actionButton3Label: GameInputLabel,
+	actionButton4Label: GameInputLabel,
+	actionButton5Label: GameInputLabel,
+	actionButton6Label: GameInputLabel,
+	specialButton1Label: GameInputLabel,
+	specialButton2Label: GameInputLabel,
```

Comment:
I am not sure if this is better as a struct or as an enumerated array since all of the fields are a `GameInputLabel`.

Reply:
I believe it would be better to just leave it as the struct to not diverge from the C documentation